debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCArrayletObjectModel_V2.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 1991, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.j9ddr.vm29.j9.gc;
+
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.vm29.types.UDATA;
+
+class GCArrayletObjectModel_V2 extends GCArrayletObjectModel_V1
```

Comment:
I thought the plan was that this would extend `GCArrayletObjectModelBase`?

Reply:
If you prefer I can update it to extend from `GCArrayletObjectModelBase`. I thought we could reuse `GCArrayletObjectModel_V1`, then we wouldn't need to reimplement the method `getTotalFootprintInBytesWithHeader`.

Reply:
If you prefer I can update it to extend from `GCArrayletObjectModelBase`. I thought we could reuse `GCArrayletObjectModel_V1`, then we wouldn't need to reimplement the method `getTotalFootprintInBytesWithHeader`.

Reply:
I think it's safer to keep the classes separate. The base class should only have implementation that's common to all versions; the two versioned classes should not be tied to each other (and `GCArrayletObjectModel_V1` may never change again).

Reply:
Updated